algebird-test/src/test/scala/com/twitter/algebird/QTreeTest.scala

```diff
@@ -68,6 +68,21 @@ class QTreeTest extends WordSpec with Matchers {
   def trueRangeSum(list: Seq[Double], from: Double, to: Double) =
     list.filter{ _ >= from }.filter{ _ < to }.sum
 
+  for (k <- Seq(3, 11, 51, 101)) {
+    s"QTree with elements (1 to $k)" should {
+      val trueMedian = (1 + k) / 2
+      s"have median $trueMedian" in {
+        implicit val sg = new QTreeSemigroup[Double](k)
```

Comment:
`QTreeSemigroup[Unit]` should work, I think, and is cheaper. People seem to be cargo-culting `QTreeSemigroup[Double]` Can we switch here?